### PR TITLE
Simplifying importing AppInsights as module, adding TS interfaces for public types

### PR DESCRIPTION
**Pull request by AlexBulankou**
- Fixing #202 
- Improving readme
- Compiling with ES5
- Improving name and tags in package.json

### CODE REVIEW

File: JavaScript/JavaScriptSDK/Context/Application.ts [hunk 1]
```diff
@@ -1,7 +1,10 @@
-﻿module Microsoft.ApplicationInsights.Context {
+﻿/// <reference path="IApplication.ts" />
+
+module Microsoft.ApplicationInsights.Context {
+
     "use strict";
 
-    export class Application {
+    export class Application implements IApplication {
         /**
```
Comment (kamilszostak) [inline JavaScript/JavaScriptSDK/Context/Application.ts]: We will need to update the definition files after those interface changes. 

My changes to DefinitelyTyped repo are in review (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/9409). 
I would wait for them to approve it and then update it to match new interfaces.

File: JavaScript/JavaScriptSDK/AppInsights.ts [hunk 2]
```diff
@@ -16,29 +17,29 @@ module Microsoft.ApplicationInsights {
...
-        autoTrackPageVisitTime: boolean;
-        disableAjaxTracking: boolean;
-        overridePageViewDuration: boolean;
-        maxAjaxCallsPerView: number;
-        disableDataLossAnalysis: boolean;
-        disableCorrelationHeaders: boolean;
-        disableFlushOnBeforeUnload: boolean;
-        enableSessionStorageBuffer: boolean;
-        cookieDomain: string;
+        instrumentationKey?: string;
```
Comment (dnduffy) [inline JavaScript/JavaScriptSDK/AppInsights.ts]: Why do strings need question marks?

File: JavaScript/JavaScriptSDK/Context/IApplication.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿module Microsoft.ApplicationInsights.Context {
+
+    "use strict";
+
+    export interface IApplication {
+        /**
+         * The application version.
+         */
+        ver: string;
```
Comment (dnduffy) [inline JavaScript/JavaScriptSDK/Context/IApplication.ts]: nitpick: I would spell out "version".

File: README.md [hunk 4]
```diff
@@ -7,16 +7,38 @@ If you don't have an Azure subscription and would like to try Application Insigh
...
+* Obtain instrumentation key from your Application Insights resource  
+* Install applicationinsights-js with npm  
+`npm install applicationinsights-js` 
+
+* Import and use it
+```
+/* import AppInsights */
+import {AppInsights, initialize as initializeAppInsights} from "applicationinsights-js"
+
+/* initialize with instrumentation key. You only need to do it once. */
```
Comment (dnduffy) [inline README.md]: nitpick: replace "it" with "this" (i.e. "...do this once.").

File: JavaScript/JavaScriptSDK/Context/IDevice.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+        /**
+         * The device language.
+         */
+        language: string;
+
+        /**
+         * The OS name.
+         */
+        os: string;
```
Comment (dnduffy) [inline JavaScript/JavaScriptSDK/Context/IDevice.ts]: nitpicks: I would spell out "operatingSystem" and/or add "Name" on the end for clarity like the one below has "version" on the end.

File: JavaScript/JavaScriptSDK/Context/IDevice.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+        /**
+         * The OS version.
+         */
+        osversion: string;
```
Comment (dnduffy) [inline JavaScript/JavaScriptSDK/Context/IDevice.ts]: nitpick: capitalize "Version".

**Comment by dnduffy** [comment]
Sync with Kamil's changes and :shipit:

File: JavaScript/JavaScriptSDK/Context/IApplication.ts (hunk 3 above)
Comment (AlexBulankou) [inline JavaScript/JavaScriptSDK/Context/IApplication.ts]: cannot change public API (unless we have a really good reason), it is already exposed to customers

File: JavaScript/JavaScriptSDK/Context/IDevice.ts (hunk 5 above)
Comment (AlexBulankou) [inline JavaScript/JavaScriptSDK/Context/IDevice.ts]: same as above

File: JavaScript/JavaScriptSDK/Context/IDevice.ts (hunk 6 above)
Comment (AlexBulankou) [inline JavaScript/JavaScriptSDK/Context/IDevice.ts]: unfortunately, same as above

**Comment by AlexBulankou** [comment]
Addressed online and offline comments, now exporting downloadAndSetup method, added unit tests

File: JavaScript/JavaScriptSDK.Tests/Selenium/Tests.html [hunk 7]
```diff
@@ -7,13 +7,28 @@
...
     <script src="http://code.jquery.com/qunit/qunit-1.18.0.js"></script>
     <script src="http://sinonjs.org/releases/sinon-1.17.2.js"></script>
+    <script src="http://cdnjs.cloudflare.com/ajax/libs/require.js/2.2.0/require.js"></script>
     <script type="text/javascript">
         window.AIResults = [];
         QUnit.testDone(function(result) {
             window.AIResults.push(result);
         });
     </script>
+    <script src="http://cdnjs.cloudflare.com/ajax/libs/require.js/2.2.0/require.js"></script>
```
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK.Tests/Selenium/Tests.html]: added twice?

Comment (AlexBulankou) [inline JavaScript/JavaScriptSDK.Tests/Selenium/Tests.html]: @MaxShehovtsov hmm, where's the second instance?

Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK.Tests/Selenium/Tests.html]: image

That's what github shows me :)

Comment (AlexBulankou) [inline JavaScript/JavaScriptSDK.Tests/Selenium/Tests.html]: Thanks! Opened PR: https://github.com/Microsoft/ApplicationInsights-JS/pull/215

**Pull request merged**